internal/serve/middleware/middleware_test.go

```diff
@@ -286,6 +288,14 @@ func Test_AuthenticateMiddleware(t *testing.T) {
 			On("GetUserID", mock.Anything, "token").
 			Return("test_user_id", nil).
 			Once()
+		mAuthManager.
+			On("GetTenantID", mock.Anything, "token").
+			Return("test_tenant_id", nil).
+			Once()
+		mTenantManager.
+			On("GetTenantByID", mock.Anything, "test_tenant_id").
+			Return(&tenant.Tenant{}, nil).
+			Once()
```

Comment:
Can we add an assert to make sure the tenant is properly saved to the context after the middleware runs ?